### ISSUE TITLE
'landmark-no-duplicate-banner' &  'landmark-no-duplicate-contentinfo' return incorrect element source

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#978]
All the elements that fail these two rules are return as related nodes unlike in the case of duplicate-id failures, where the first matching node is returned as a failure node and the rest of the nodes are returned as related nodes 

![image](https://user-images.githubusercontent.com/22068434/41998784-6c399ae6-7a10-11e8-8a16-fe0399ed9de7.png)

<pre><code>axe-core version: 3.0.3
axe-Coconut version: 3.4.0-alpha.2</code></pre>

**Comment by WilcoFiers** [comment]
This is the same issue as #878.

**Comment by WilcoFiers** [comment]
Proposal to use the after function to turn the related nodes into matched nodes, so that the result would go from look like this:

```js
[{
   target: 'html'
   relatedNode: [{
     target: '#div1'
   }, {
     target: '#div2'
   }, {
     target: '#div3'
   }]
}]
```

to looking like this:

```js
[{
  target: '#div1',
  relatedNodes: [{
    target: 'html'
  }]
}, {
  target: '#div2',
  relatedNodes: [{
    target: 'html'
  }]
}, {
  target: '#div3',
  relatedNodes: [{
    target: 'html'
  }]
}]
```

**Comment by marcysutton** [comment]
We also talked about limiting results to common ancestors, so we don't get a bunch of extra noise. This will take the region rule from one result to many, to make it easier to understand.

**Comment by WilcoFiers** [comment]
Me and Jey have been looking into this, and it isn't as simple as tweaking the data in "after". Unfortunately axe-core doesn't let you flip related nodes and selected nodes. We'd have to rewrite these rules to do this I think, which is a bummer. I do think that's probably the right way to do it.

**Comment by WilcoFiers** [comment]
Talking to Steve about this. The way to deal with this is to select all the landmarks in the rule, and use the after method to remove any duplicates after the first. This would be similar to how we do it in #1787 and #1649.

**Comment by jeankaplansky** [comment]
new rule landmark-no-duplicate-main infrastructure created 1/27/2020

**Comment by jeankaplansky** [comment]
landmark-no-duplicate-main rule create in axe-core rule help and release notes 2/03/2020

**Comment by padmavemulapati** [comment]
Ensuring no duplicate landmarks , verified with the latest axe.
![image](https://user-images.githubusercontent.com/47409579/74035235-e8c54b00-49df-11ea-93b4-adc577022c75.png)
